inquire/src/ui/backend.rs

```diff
@@ -695,6 +697,8 @@ pub mod date {
                         }
                     } else if date_it == today {
                         style_sheet = self.render_config.calendar.today_date;
+                    } else if is_weekend(date_it) {
```

Comment:
It won't make a difference when both colors are the same, but in the case they're different, I think the (day in the prev/next month) style should have precedence over the (day is weekend) style.

Like in this example:
<img width="176" alt="image" src="https://github.com/mikaelmello/inquire/assets/17488630/4d27f130-7196-481e-8393-fd9b46febd74">

<img width="182" alt="image" src="https://github.com/mikaelmello/inquire/assets/17488630/883abc00-7369-4ac4-98ce-34ebeaa118c7">

I feel the second screenshot represents a better UX when parsing the calendar